geopandas/array.py

```diff
@@ -1224,23 +1224,30 @@ def _formatter(self, boxed=False):
 
             precision = geopandas.options.display_precision
             if precision is None:
-                # dummy heuristic based on 10 first geometries that should
-                # work in most cases
-                with warnings.catch_warnings():
-                    warnings.simplefilter("ignore", category=RuntimeWarning)
-                    xmin, ymin, xmax, ymax = self[~self.isna()][:10].total_bounds
-                if (
-                    (-180 <= xmin <= 180)
-                    and (-180 <= xmax <= 180)
-                    and (-90 <= ymin <= 90)
-                    and (-90 <= ymax <= 90)
-                ):
-                    # geographic coordinates
-                    precision = 5
+                if self.crs:
+                    if self.crs.is_projected:
+                        precision = 3
+                    else:
+                        precision = 5
```

Comment:
It seems that we don't have this option tested currently. Can you add one simple test for `__repr__` based on geometry in geographic projection (4326)?

Reply:
Here https://github.com/geopandas/geopandas/blob/2a5c42f8fcf7093caf5e59c012ed05817b68569f/geopandas/tests/test_pandas_methods.py#L42

Reply:
Sure thing! What does the 4326 means?

Reply:
Actually, it would be good to explicitly check both options based on CRS since the existing tests all fall to heuristic.

Reply:
Should I add a test for the case when the crs is projected?

Reply:
Yes. Two tests are needed. One with some explicit projected CRS and one with explicit geographic CRS.